pkg/client/core/service_application_client.go

```diff
@@ -127,6 +113,165 @@ func (s *applicationService) CreateArgoCDApplicationManifest(opts client.CreateA
 	return nil
 }
 
+// DeleteArgoCDApplicationManifest removes necessary files related to the ArgoCD integration
+// This function adds a finalizer to the ArgoCD application manifest before deletion, ensuring a cascading delete.
+// https://argo-cd.readthedocs.io/en/stable/user-guide/app_deletion/#about-the-deletion-finalizer
+func (s *applicationService) DeleteArgoCDApplicationManifest(opts client.DeleteArgoCDApplicationManifestOpts) error {
+	relativeArgoCDApplicationManifestPath := getRelativeArgoCDManifestPath(opts.Cluster, opts.Application)
+	absoluteArgoCDApplicationManifestPath := path.Join(s.absoluteRepositoryDir, relativeArgoCDApplicationManifestPath)
+
+	err := s.deleteFileInRemote(opts.Cluster, opts.Application, relativeArgoCDApplicationManifestPath)
+	if err != nil {
+		return fmt.Errorf("deleting application manifest in repository: %w", err)
+	}
+
+	err = s.fs.Remove(absoluteArgoCDApplicationManifestPath)
+	if err != nil {
+		if stderrors.Is(err, os.ErrNotExist) {
+			return nil
+		}
+
+		return errors.E(err, "removing ArgoCD application manifest")
+	}
+
+	err = s.patchArgoCDApplicationManifest(opts.Application.Metadata.Name)
+	if err != nil {
+		if stderrors.Is(err, kubectl.ErrNotFound) {
+			return nil
+		}
+
+		return fmt.Errorf("adding finalizer to application manifest: %w", err)
+	}
+
+	manifest, err := scaffold.GenerateArgoCDApplicationManifest(scaffold.GenerateArgoCDApplicationManifestOpts{
+		Name:          opts.Application.Metadata.Name,
+		Namespace:     opts.Application.Metadata.Namespace,
+		IACRepoURL:    opts.Cluster.Github.URL(),
+		SourceSyncDir: getRelativeOverlayDirectory(opts.Cluster, opts.Application),
+	})
+	if err != nil {
+		return fmt.Errorf("generating ArgoCD application manifest: %w", err)
+	}
+
+	err = s.kubectl.Delete(manifest)
+	if err != nil {
+		return fmt.Errorf("deleting ArgoCD application manifest from cluster: %w", err)
+	}
+
+	return nil
+}
+
+// HasArgoCDIntegration checks if an application has been set up with ArgoCD
+func (s *applicationService) HasArgoCDIntegration(_ context.Context, opts client.HasArgoCDIntegrationOpts) (bool, error) {
+	absoluteArgoCDApplicationManifestPath := path.Join(
+		s.absoluteRepositoryDir,
+		getRelativeArgoCDManifestPath(opts.Cluster, opts.Application),
+	)
+
+	_, err := s.fs.Stat(absoluteArgoCDApplicationManifestPath)
+	if err != nil {
+		if stderrors.Is(err, os.ErrNotExist) {
+			return false, nil
+		}
+
+		return false, errors.E(err, "checking existence of ArgoCD application manifest: %w", err)
+	}
+
+	return true, nil
+}
+
+// deleteFileInRemote creates a commit on the main branch of a repository that removes a file path
```

Comment:
not only creates, but also pushes the commit

Reply:
You can't create a commit on a remote branch without actually pushing it :D but I'll rephrase